src/nvim/ex_docmd.c

```diff
@@ -9952,3 +9953,104 @@ bool cmd_can_preview(char_u *cmd)
 
   return false;
 }
+
+/// Return the list of the commands in the specified buffer,
+/// if no buffer is specified, then return list of global(non-buffer) the
+/// commands
+///
+/// @param[in] buf the specified buffer, if NULL then global commands required
+///
+/// @return The array of the commands and the information of each command
+ArrayOf(Dictionary) get_commands(buf_T *buf)
+{
+  Array rv = ARRAY_DICT_INIT;
+  int i = 0, j = 0;
+  char Arg[2] = "";
+  garray_T *gap;
+  if (buf == NULL) {
+    gap = &ucmds;
+  } else {
+    gap = &buf->b_ucmds;
+  }
+  for (i = 0; i < gap->ga_len; i++) {
+    Dictionary dic = ARRAY_DICT_INIT;
+    bool ck = false, ck2 = false;
+    char Range[10] = "";
+    ucmd_T *cmd = USER_CMD_GA(gap, i);
+
+    // Name
+    PUT(dic, "Name", STRING_OBJ(cstr_to_string((const char *)cmd->uc_name)));
+
+    // Argument
+    switch (cmd->uc_argt & (EXTRA|NOSPC|NEEDARG)) {
+    case 0:                    Arg[0] = '0'; break;
+    case(EXTRA):               Arg[0] = '*'; break;
+    case(EXTRA|NOSPC):         Arg[0] = '?'; break;
+    case(EXTRA|NEEDARG):       Arg[0] = '+'; break;
+    case(EXTRA|NOSPC|NEEDARG): Arg[0] = '1'; break;
+    }
+    PUT(dic, "Argument", STRING_OBJ(cstr_to_string((const char *)Arg)));
+
+    // Defination
+    PUT(dic, "Defination",
+        STRING_OBJ(cstr_to_string((const char *)cmd->uc_rep)));
+
+    // Complete
+    for (j = 0; command_complete[j].expand != 0; j++) {
+      if (cmd->uc_compl == command_complete[j].expand) {
+        PUT(dic, "Complete",
+            STRING_OBJ(cstr_to_string(command_complete[j].name)));
+        ck = true;
+        break;
+      }
+    }
+    if (!ck) {
+      PUT(dic, "Complete",
+          STRING_OBJ(cstr_to_string("")));
+    }
+    // Range
+    if (cmd->uc_argt & (RANGE|COUNT)) {
+      if (cmd->uc_argt & COUNT) {
+        // -count=N
+        snprintf((char *)Range, sizeof(Range), "%" PRId64 "c",
+                 (int64_t)cmd->uc_def);
+      } else if (cmd->uc_argt & DFLALL) {
+        Range[0] = '%';
+      } else if (cmd->uc_def >= 0) {
+        // -range=N
+        snprintf((char *)Range, sizeof(Range), "%" PRId64 "",
+                 (int64_t)cmd->uc_def);
+      } else {
+        Range[0] = '.';
+      }
+    } else {
+      Range[0] = '\0';
+    }
+    PUT(dic, "Range",
+        STRING_OBJ(cstr_to_string((const char *)Range)));
+
+    // Address
+    for (j = 0; addr_type_complete[j].expand != -1; j++) {
+      if (addr_type_complete[j].expand != ADDR_LINES
+          && addr_type_complete[j].expand == cmd->uc_addr_type) {
+        PUT(dic, "Address",
+            STRING_OBJ(cstr_to_string(addr_type_complete[j].name)));
+        ck2 = true;
+        break;
+      }
+    }
+    if (!ck2) {
+      PUT(dic, "Address",
+          STRING_OBJ(cstr_to_string("")));
+    }
+
+    // ScriptID
+    PUT(dic, "ScriptID", INTEGER_OBJ(cmd->uc_scriptID));
+
+    // Compl Arg
+    PUT(dic, "Compl Arg",
```

Comment:
key names never have whitespace and are never capitalized.

Reply:
Was thinking of Compl Arg same, didn't get the feel of a coder here, I change it